### PR TITLE
[bitnami/postgres-ha] Enable init-chmod-data when using extendedConf

### LINKED ISSUE #1961: [bitnami/postgresql-ha] extendedConf breaks permissions for file: postgresql.conf
**Which chart**:

bitnami/postgresql-ha version 1.4.4, app version 11.7.0

**Description**

I'm using helm chart  with configuration:
```
postgresql:
  replicaCount: 2
  username: postgres
  password: ****
  repmgrPassword: ***
  extendedConf:
    shared_preload_libraries: repmgr
volumePermissions:
  enabled: false
postgresqlImage:
  debug: true
pgpool:
  replicaCount: 2
  pdb:
    create: false
service:
  type: LoadBalancer
  loadBalancerIP: 35.*.*.*
ingress:
  enabled: false
persistence:
  enabled: true
  storageClass: ssd
```
and receive output 
```
postgresql 13:26:33.63 INFO  ==> postgresql.conf file not detected. Generating it...
cp: cannot create regular file '/bitnami/postgresql/conf/postgresql.conf': Permission denied
postgresql 13:26:34.04 INFO  ==> Stopping PostgreSQL...
```
but when remove this parameter from configuration file

```
  extendedConf:
    shared_preload_libraries: repmgr
```

everything is working fine

```
repmgr 13:28:15.72 INFO  ==> Preparing PostgreSQL configuration...
repmgr 13:28:15.73 DEBUG ==> Injecting a new postgresql.conf file...
postgresql 13:28:15.73 INFO  ==> postgresql.conf file not detected. Generating it...
repmgr 13:28:16.22 DEBUG ==> Injecting a new pg_hba.conf file...
```

P.S. each time i'm using fresh PVC and helm install

**Steps to reproduce the issue:**

1. enable parameter `extendedConf` 
2. `helm install postgresql bitnami/postgresql-ha --namespace postgresql --values ./production.yaml`

**Describe the results you received:**

```
postgresql 13:26:33.63 INFO  ==> postgresql.conf file not detected. Generating it...
cp: cannot create regular file '/bitnami/postgresql/conf/postgresql.conf': Permission denied
postgresql 13:26:34.04 INFO  ==> Stopping PostgreSQL...
```

**Describe the results you expected:**

```
repmgr 13:57:28.27 INFO  ==> Preparing PostgreSQL configuration...
repmgr 13:57:28.27 DEBUG ==> Injecting a new postgresql.conf file...
postgresql 13:57:28.28 INFO  ==> postgresql.conf file not detected. Generating it...
repmgr 13:57:28.34 DEBUG ==> Injecting a new pg_hba.conf file...
repmgr 13:57:28.35 INFO  ==> Preparing repmgr configuration...
repmgr 13:57:28.35 DEBUG ==> Node ID: 1000, Rol: primary, Primary Node: 
repmgr 13:57:28.36 INFO  ==> Initializing Repmgr...
postgresql 13:57:28.36 INFO  ==> Initializing PostgreSQL database...
postgresql 13:57:28.36 DEBUG ==> Copying files from /bitnami/postgresql/conf to /opt/bitnami/postgresql/conf
postgresql 13:57:28.37 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql 13:57:28.37 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
postgresql 13:57:28.38 DEBUG ==> Ensuring expected directories/files exist...
postgresql 13:57:28.39 INFO  ==> Deploying PostgreSQL with persisted data...

```

**Additional information you deem important (e.g. issue happens only occasionally):**

key `shared_preload_libraries` will be used to add other extensions when it will be fixed (for example repmgr,timescaledb)

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.0.0", GitCommit:"e29ce2a54e96cd02ccfce88bee4f58bb6e2a28b6", GitTreeState:"clean", GoVersion:"go1.13.4"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-gke.8", GitCommit:"a9973cbb2722793e2ea08d20880633ca61d3e669", GitTreeState:"clean", BuildDate:"2020-02-07T00:50:57Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```

### LINKED ISSUE #6197: [bitnami/postgresql-ha] postgresql.extendedConf requires volumePermissions.enabled to work properly
**Which chart**:
- name: postgresql-ha
  repository: https://charts.bitnami.com/bitnami
  version: 6.9.0

**Describe the bug**
This is a regression of #1961
While #2140 fixed #1961, initContainers init-chmod-data requires volumePermissions.enabled to be truly, see https://github.com/bitnami/charts/blob/master/bitnami/postgresql-ha/templates/postgresql/statefulset.yaml#L73 . But this behavior is nether documented nor corrected in the chart. 

**To Reproduce**

Deploy postgresql-ha with non-empty postgresql.extendedConf and volumePermissions.enabled=false, and monitor the -postgresql-ha-postgresql-0 pod. It will crash with the same error in #1961 .

**Expected behavior**

postgresql.conf file generates correctly in the first run.

**Version of Helm and Kubernetes**:

<details>

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"clean", GoVersion:"go1.13.12"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"c96aede7b5205121079932896c4ad89bb93260af", GitTreeState:"clean", BuildDate:"2020-06-17T11:41:22Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"clean", BuildDate:"2021-03-18T01:02:01Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
N/A
</details>

### PR DESCRIPTION
**Pull request by Librazy**
**Description of the change**

Fix https://github.com/bitnami/charts/issues/6197 by enabling init-chmod-data initContainer when `postgresql.extendedConf` is not empty.

**Benefits**

`postgresql.extendedConf` works out of box and won't fails surprisingly with `'/bitnami/postgresql/conf/postgresql.conf': Permission denied` 

**Possible drawbacks**

An initContainer increase startup time and may requires a extra image. I don't know whether there is a better way to fix #1961 without initContainers.

**Applicable issues**

  - fixes #6197

**Additional information**

N/A

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Review by juan131 — CHANGES_REQUESTED** [review]
I don't see any reason to enable the init container when using the "extended configuration" feature". 

This feature mounts a ConfigMap (read only) at `/bitnami/postgresql/conf/conf.d/`. Then, the configuration file (**postgresql.conf**) includes the directive below to include the extended configuration:

```
include_dir = 'conf.d'
```

See the volume mount points:

```
          volumeMounts:
            - name: postgresql-extended-config
              mountPath: /bitnami/postgresql/conf/conf.d/
            - name: data
              mountPath: /bitnami/postgresql
```

If you're installing the chart in a cluster with support for Security Context(s), the default securityContext should adapt the permissions in the data directory, and you shouldn't find any permissions issue. See:

```
spec:
  template:
    spec:
      securityContext:
        fsGroup: 1001
```

**Comment by Librazy** [comment]
This is how https://github.com/bitnami/charts/issues/1961 was fixed by @marcosbc 's https://github.com/bitnami/charts/pull/2140

**Comment by Mauraza** [comment]
Hi @Librazy, 

Could you please test if the solution proposed by @juan131 works?

**Comment by Librazy** [comment]
You can take a look at https://github.com/bitnami/charts/pull/2140 for reference.

My cluster supports fsGroup:
```yaml
# kubectl version -o yaml
clientVersion:
  buildDate: "2020-06-17T11:41:22Z"
  compiler: gc
  gitCommit: c96aede7b5205121079932896c4ad89bb93260af
  gitTreeState: clean
  gitVersion: v1.18.4
  goVersion: go1.13.9
  major: "1"
  minor: "18"
  platform: linux/amd64
serverVersion:
  buildDate: "2020-11-12T09:11:15Z"
  compiler: gc
  gitCommit: 7cd5e9086de8ae25d6a1514d0c87bac67ca4a481
  gitTreeState: clean
  gitVersion: v1.18.12
  goVersion: go1.13.15
  major: "1"
  minor: "18"
  platform: linux/amd64
```
## Let's install with extendedConf and disabled volumePermissions container
```yaml
# cat values.yml 
postgresql:
  extendedConf: |
    shared_buffers = 256MB
# volumePermissions:
#   enabled: true
```
```
# helm install pg ./postgresql-ha-6.9.1.tgz -f values.yml 
NAME: pg
LAST DEPLOYED: Wed Apr 28 05:25:12 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
** Please be patient while the chart is being deployed **

PostgreSQL can be accessed through Pgpool via port 5432 on the following DNS name from within your cluster:

    pg-postgresql-ha-pgpool.default.svc.cluster.local

Pgpool acts as a load balancer for PostgreSQL and forward read/write connections to the primary node while read-only connections are forwarded to standby nodes.

To get the password for "postgres" run:

    export POSTGRES_PASSWORD=$(kubectl get secret --namespace default pg-postgresql-ha-postgresql -o jsonpath="{.data.postgresql-password}" | base64 --decode)

To get the password for "repmgr" run:

    export REPMGR_PASSWORD=$(kubectl get secret --namespace default pg-postgresql-ha-postgresql -o jsonpath="{.data.repmgr-password}" | base64 --decode)

To connect to your database run the following command:

    kubectl run pg-postgresql-ha-client --rm --tty -i --restart='Never' --namespace default --image docker.io/bitnami/postgresql-repmgr:11.11.0-debian-10-r58 --env="PGPASSWORD=$POSTGRES_PASSWORD"  \
        --command -- psql -h pg-postgresql-ha-pgpool -p 5432 -U postgres -d postgres

To connect to your database from outside the cluster execute the following commands:

    kubectl port-forward --namespace default svc/pg-postgresql-ha-pgpool 5432:5432 &
    psql -h 127.0.0.1 -p 5432 -U postgres -d postgres
# kubectl logs -f pg-postgresql-ha-postgresql-0
postgresql-repmgr 05:25:29.39 
postgresql-repmgr 05:25:29.40 Welcome to the Bitnami postgresql-repmgr container
postgresql-repmgr 05:25:29.40 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
postgresql-repmgr 05:25:29.40 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
postgresql-repmgr 05:25:29.40 
postgresql-repmgr 05:25:29.42 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
postgresql-repmgr 05:25:29.44 INFO  ==> Validating settings in REPMGR_* env vars...
postgresql-repmgr 05:25:29.45 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql-repmgr 05:25:29.45 INFO  ==> Querying all partner nodes for common upstream node...
postgresql-repmgr 05:25:29.53 INFO  ==> There are no nodes with primary role. Assuming the primary role...
postgresql-repmgr 05:25:29.53 INFO  ==> Preparing PostgreSQL configuration...
postgresql-repmgr 05:25:29.54 INFO  ==> postgresql.conf file not detected. Generating it...
cp: cannot create regular file '/bitnami/postgresql/conf/postgresql.conf': Permission denied
# kubectl logs -f pg-postgresql-ha-postgresql-1
postgresql-repmgr 05:25:28.96 
postgresql-repmgr 05:25:28.96 Welcome to the Bitnami postgresql-repmgr container
postgresql-repmgr 05:25:28.96 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
postgresql-repmgr 05:25:28.96 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
postgresql-repmgr 05:25:28.96 
postgresql-repmgr 05:25:28.98 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
postgresql-repmgr 05:25:29.00 INFO  ==> Validating settings in REPMGR_* env vars...
postgresql-repmgr 05:25:29.01 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql-repmgr 05:25:29.01 INFO  ==> Querying all partner nodes for common upstream node...
postgresql-repmgr 05:25:29.05 INFO  ==> Preparing PostgreSQL configuration...
postgresql-repmgr 05:25:29.06 INFO  ==> postgresql.conf file not detected. Generating it...
cp: cannot create regular file '/bitnami/postgresql/conf/postgresql.conf': Permission denied
```
## When we enable volumePermissions container, it works fine.
```yaml
# cat values.yml 
postgresql:
  extendedConf: |
    shared_buffers = 256MB
volumePermissions:
  enabled: true
```
```
# helm install pg ./postgresql-ha-6.9.1.tgz -f values.yml 
# kubectl logs -f pg-postgresql-ha-postgresql-0
postgresql-repmgr 05:30:53.12 
postgresql-repmgr 05:30:53.12 Welcome to the Bitnami postgresql-repmgr container
postgresql-repmgr 05:30:53.12 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
postgresql-repmgr 05:30:53.12 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
postgresql-repmgr 05:30:53.13 
postgresql-repmgr 05:30:53.14 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
postgresql-repmgr 05:30:53.17 INFO  ==> Validating settings in REPMGR_* env vars...
postgresql-repmgr 05:30:53.17 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql-repmgr 05:30:53.17 INFO  ==> Querying all partner nodes for common upstream node...
postgresql-repmgr 05:30:53.21 INFO  ==> There are no nodes with primary role. Assuming the primary role...
postgresql-repmgr 05:30:53.22 INFO  ==> Preparing PostgreSQL configuration...
postgresql-repmgr 05:30:53.22 INFO  ==> postgresql.conf file not detected. Generating it...
postgresql-repmgr 05:30:53.32 INFO  ==> Preparing repmgr configuration...
postgresql-repmgr 05:30:53.32 INFO  ==> Initializing Repmgr...
postgresql-repmgr 05:30:53.33 INFO  ==> Initializing PostgreSQL database...
postgresql-repmgr 05:30:53.34 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql-repmgr 05:30:53.34 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
postgresql-repmgr 05:30:54.94 INFO  ==> Starting PostgreSQL in background...
postgresql-repmgr 05:30:55.07 INFO  ==> Changing password of postgres
postgresql-repmgr 05:30:55.09 INFO  ==> Creating replication user repmgr
postgresql-repmgr 05:30:55.12 INFO  ==> Configuring replication parameters
postgresql-repmgr 05:30:55.15 INFO  ==> Configuring fsync
postgresql-repmgr 05:30:55.16 INFO  ==> Starting PostgreSQL in background...
postgresql-repmgr 05:30:55.17 INFO  ==> Creating repmgr user: repmgr
postgresql-repmgr 05:30:55.21 INFO  ==> Creating repmgr database: repmgr
postgresql-repmgr 05:30:55.62 INFO  ==> Stopping PostgreSQL...
waiting for server to shut down.... done
server stopped
postgresql-repmgr 05:30:55.83 INFO  ==> Starting PostgreSQL in background...
postgresql-repmgr 05:30:55.95 INFO  ==> Registering Primary...
postgresql-repmgr 05:30:56.09 INFO  ==> Loading custom scripts...
postgresql-repmgr 05:30:56.10 INFO  ==> Stopping PostgreSQL...
waiting for server to shut down.... done
server stopped
postgresql-repmgr 05:30:56.31 INFO  ==> ** PostgreSQL with Replication Manager setup finished! **

postgresql-repmgr 05:30:56.34 INFO  ==> Starting PostgreSQL in background...
waiting for server to start....2021-04-28 05:30:56.382 GMT [257] LOG:  pgaudit extension initialized
2021-04-28 05:30:56.382 GMT [257] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2021-04-28 05:30:56.382 GMT [257] LOG:  listening on IPv6 address "::", port 5432
2021-04-28 05:30:56.387 GMT [257] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2021-04-28 05:30:56.429 GMT [257] LOG:  redirecting log output to logging collector process
2021-04-28 05:30:56.429 GMT [257] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
2021-04-28 05:30:56.434 GMT [259] LOG:  database system was shut down at 2021-04-28 05:30:56 GMT
2021-04-28 05:30:56.439 GMT [257] LOG:  database system is ready to accept connections
 done
server started
postgresql-repmgr 05:30:56.46 INFO  ==> ** Starting repmgrd **
[2021-04-28 05:30:56] [NOTICE] repmgrd (repmgrd 5.2.1) starting up
INFO:  set_repmgrd_pid(): provided pidfile is /opt/bitnami/repmgr/tmp/repmgr.pid
[2021-04-28 05:30:56] [NOTICE] starting monitoring of node "pg-postgresql-ha-postgresql-0" (ID: 1000)
[2021-04-28 05:30:56] [NOTICE] monitoring cluster primary "pg-postgresql-ha-postgresql-0" (ID: 1000)
2021-04-28 05:31:09.246 GMT [285] WARNING:  pg_getnameinfo_all() failed: Temporary failure in name resolution
```

## Explorering what happened when volumePermissions is disabled

```yml
# cat values.yml 
postgresql:
  command: 
    - bash
  args:
    - -c
    - "ls -laR /bitnami/postgresql/"
  extendedConf: |
    shared_buffers = 256MB
# volumePermissions:
#   enabled: true
```
```
# kubectl logs -f pg-postgresql-ha-postgresql-0
/bitnami/postgresql/:
total 28
drwxrwsr-x 4 root 1001  4096 Apr 28 05:36 .
drwxr-xr-x 3 root root  4096 Apr 20 04:52 ..
drwxr-sr-x 3 root 1001  4096 Apr 28 05:36 conf
drwxrws--- 2 root 1001 16384 Apr 28 05:36 lost+found

/bitnami/postgresql/conf:
total 12
drwxr-sr-x 3 root 1001 4096 Apr 28 05:36 .
drwxrwsr-x 4 root 1001 4096 Apr 28 05:36 ..
drwxrwsrwx 3 root 1001 4096 Apr 28 05:36 conf.d

/bitnami/postgresql/conf/conf.d:
total 12
drwxrwsrwx 3 root 1001 4096 Apr 28 05:36 .
drwxr-sr-x 3 root 1001 4096 Apr 28 05:36 ..
drwxr-sr-x 2 root 1001 4096 Apr 28 05:36 ..2021_04_28_05_36_31.830981167
lrwxrwxrwx 1 root root   31 Apr 28 05:36 ..data -> ..2021_04_28_05_36_31.830981167
lrwxrwxrwx 1 root root   20 Apr 28 05:36 override.conf -> ..data/override.conf

/bitnami/postgresql/conf/conf.d/..2021_04_28_05_36_31.830981167:
total 12
drwxr-sr-x 2 root 1001 4096 Apr 28 05:36 .
drwxrwsrwx 3 root 1001 4096 Apr 28 05:36 ..
-rw-r--r-- 1 root 1001   22 Apr 28 05:36 override.conf

/bitnami/postgresql/lost+found:
total 20
drwxrws--- 2 root 1001 16384 Apr 28 05:36 .
drwxrwsr-x 4 root 1001  4096 Apr 28 05:36 ..
```

## And when enabled.

``` yaml
# cat values.yml 
postgresql:
  command: 
    - bash
  args:
    - -c
    - "ls -laR /bitnami/postgresql/"
  extendedConf: |
    shared_buffers = 256MB
volumePermissions:
  enabled: true
```
```
# kubectl logs -f pg-postgresql-ha-postgresql-0
/bitnami/postgresql/:
total 36
drwxrwsr-x 6 root 1001  4096 Apr 28 05:46 .
drwxr-xr-x 3 root root  4096 Apr 20 04:52 ..
drwx--S--- 3 1001 1001  4096 Apr 28 05:46 conf
drwx--S--- 2 1001 1001  4096 Apr 28 05:46 data
drwx--S--- 2 1001 1001  4096 Apr 28 05:46 lock
drwxrws--- 2 root 1001 16384 Apr 28 05:46 lost+found

/bitnami/postgresql/conf:
total 12
drwx--S--- 3 1001 1001 4096 Apr 28 05:46 .
drwxrwsr-x 6 root 1001 4096 Apr 28 05:46 ..
drwxrwsrwx 3 root 1001 4096 Apr 28 05:46 conf.d

/bitnami/postgresql/conf/conf.d:
total 12
drwxrwsrwx 3 root 1001 4096 Apr 28 05:46 .
drwx--S--- 3 1001 1001 4096 Apr 28 05:46 ..
drwxr-sr-x 2 root 1001 4096 Apr 28 05:46 ..2021_04_28_05_46_36.874338412
lrwxrwxrwx 1 root root   31 Apr 28 05:46 ..data -> ..2021_04_28_05_46_36.874338412
lrwxrwxrwx 1 root root   20 Apr 28 05:46 override.conf -> ..data/override.conf

/bitnami/postgresql/conf/conf.d/..2021_04_28_05_46_36.874338412:
total 12
drwxr-sr-x 2 root 1001 4096 Apr 28 05:46 .
drwxrwsrwx 3 root 1001 4096 Apr 28 05:46 ..
-rw-r--r-- 1 root 1001   22 Apr 28 05:46 override.conf

/bitnami/postgresql/data:
total 8
drwx--S--- 2 1001 1001 4096 Apr 28 05:46 .
drwxrwsr-x 6 root 1001 4096 Apr 28 05:46 ..

/bitnami/postgresql/lock:
total 8
drwx--S--- 2 1001 1001 4096 Apr 28 05:46 .
drwxrwsr-x 6 root 1001 4096 Apr 28 05:46 ..

/bitnami/postgresql/lost+found:
total 20
drwxrws--- 2 root 1001 16384 Apr 28 05:46 .
drwxrwsr-x 6 root 1001  4096 Apr 28 05:46 ..
```

## When both is disabled

```yaml
# cat values.yml 
postgresql:
  command: 
    - bash
  args:
    - -c
    - "ls -laR /bitnami/postgresql/"
#   extendedConf: |
#     shared_buffers = 256MB
# volumePermissions:
#   enabled: true
```
```
# kubectl logs -f pg-postgresql-ha-postgresql-0
/bitnami/postgresql/:
total 24
drwxrwsr-x 3 root 1001  4096 Apr 28 05:50 .
drwxr-xr-x 3 root root  4096 Apr 20 04:52 ..
drwxrws--- 2 root 1001 16384 Apr 28 05:50 lost+found

/bitnami/postgresql/lost+found:
total 20
drwxrws--- 2 root 1001 16384 Apr 28 05:50 .
drwxrwsr-x 3 root 1001  4096 Apr 28 05:50 ..
```

It seems that /bitnami/postgresql/conf is not writeable by 1001 when volumePermissions is disabled and extendedConf enabled.

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 1]
```diff
@@ -70,7 +70,7 @@ spec:
       serviceAccountName: {{ default (include "common.names.fullname" . ) .Values.serviceAccount.name}}
       {{- end }}
       initContainers:
-      {{- if and .Values.volumePermissions.enabled .Values.persistence.enabled }}
+      {{- if and .Values.persistence.enabled (or (or (not (empty .Values.postgresql.extendedConf)) (not (empty .Values.postgresql.extendedConfCM)) ) .Values.volumePermissions.enabled) }}
```
Comment (juan131) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:73]: We can make this simpler. Please Try with:
```suggestion
      {{- if and .Values.persistence.enabled (or (or .Values.postgresql.extendedConf .Values.postgresql.extendedConfCM) .Values.volumePermissions.enabled) }}
```

**Review by juan131 — COMMENTED** [review]
I understands what happens here...

The `/bitnami/postgresql/conf` directory exists by default in the image with the right permissions. See:

```console
$ docker run --rm -it bitnami/postgresql-repmgr:11 ls -la /bitnami/postgresql
total 12
drwxrwxr-x 3 0 postgres 4096 Apr 29 00:33 .
drwxr-xr-x 3 0 postgres 4096 Apr 29 00:33 ..
drwxrwxr-x 3 0 postgres 4096 Apr 29 00:33 conf
```

That said, when mounting the data volume (which is empty by default) at `/bitnami/postgresl` it's removed. Then, when mounting the "extendedConf" ConfigMap at `/bitnami/postgresl/conf/conf.d` it's created **BUT without writing permissions for the "group".**

As a consequence, the permissions **must** be adapted using the init container.

**Review by juan131 — APPROVED** [review]
LGTM

File: bitnami/postgresql-ha/Chart.yaml [hunk 2]
```diff
@@ -27,4 +27,4 @@ name: postgresql-ha
 sources:
   - https://github.com/bitnami/bitnami-docker-postgresql
   - https://www.postgresql.org/
-version: 7.0.0
+version: 7.1.0
```
Comment (Mauraza) [inline bitnami/postgresql-ha/Chart.yaml:30]: change the version to 7.2.0

**Review by Mauraza — COMMENTED** [review]
Hi @Librazy,

could you update the version?

**Comment by Librazy** [comment]
Sorry for the delay and chart version is now updated

**Comment by acourouppe** [comment]
Hi @Librazy, 
Why it is it required to enable persistence to be able to use extendedConf?
I mean, I'd like to use the extend conf but in dev we don't want to enable data persistence. With the current implementation, it looks like I'm forced to enable the persistence in order  for .Values.volumePermissions.enabled to be taken into account and the init-chmod-data container to be run.  Here is the condition. 

`{{- if and .Values.persistence.enabled (or (or (not (empty .Values.postgresql.extendedConf)) (not (empty .Values.postgresql.extendedConfCM)) ) .Values.volumePermissions.enabled) }}`

**Comment by juan131** [comment]
Hi @acourouppe 

AFAIK the init-chmod container isn't required if you're disabling persistence since the scenario I explained in [this comment](https://github.com/bitnami/charts/pull/6206#pullrequestreview-647910293) shouldn't be reproduced.

Could you please share the **values.yaml** you're using so I can reproduce the issue on my side? Please also share the chart version you're using.

**Comment by acourouppe** [comment]
Hi @juan131 
I'm using Helm chart 8.1.3 with postgresqlImage docker image overridden to 13.5.0-debian-10-r35.

Actually from my observations: 
* extendedConf + volumePermissions=disabled +  persistence=disabled => exception - cp: cannot create regular file '/bitnami/postgresql/conf/postgresql.conf': Permission denied
* extendedConf + volumePermissions=enabled +  persistence=disabled =>  exception -cp: cannot create regular file '/bitnami/postgresql/conf/postgresql.conf': Permission denied
* extendedConf + volumePermissions=enabled + persistence=enabled => works
* extendedConf + volumePermissions=disabled + persistence=enabled => works as well...

My values file looks like this: 
```
postgresql-ha:
  enabled: true
  metrics:
    enabled:
      true
  persistence:
    enabled: true
  volumePermissions:
    enabled: true
  pgpool:
    replicaCount: 1
    customUsers:
      usernames: opf16_0_postgres
      passwords: password
  postgresql:
    password: password
    repmgrPassword: password
    database: bph
    replicaCount: 1
    initContainers:
      - name: bph-custom-init-scripts
        image: opf-docker.coastguard.clear2pay.com/test/postgres-test-opf:${project.version}
        command: ["/bin/sh", "-c", "unzip /tmp/bph-dbscripts.zip -d /tmp/${docker-postgres-schema} && chown -R 1001 /tmp/${docker-postgres-schema}"]
        volumeMounts:
          - name: bph-custom-init-scripts
            mountPath: /tmp/${docker-postgres-schema}
    extraVolumeMounts:
      - name: bph-custom-init-scripts
        mountPath: /tmp/${docker-postgres-schema}
      - name: bph-custom-init-scripts
        mountPath: /docker-entrypoint-initdb.d/startUpdateDb.sh
        subPath: startUpdateDb.sh
    extraVolumes:
      - name: bph-custom-init-scripts
        emptyDir: {}
    extraEnvVars:
      - name: "DB_UPDATE_OPTS"
        value: "-t"
    extendedConf: |-
      listen_addresses = '*'
      max_prepared_transactions = 400
      max_connections = 400
```

Here is the exception I get
```
postgresql postgresql-repmgr 12:53:59.50 DEBUG ==> Skipping: failed to get primary from the node 'ci-postgresql-ha-postgresql-0.ci-postgresql-ha-postgresql-headless.opf.svc.cluster.local │
│ postgresql postgresql-repmgr 12:53:59.51 INFO  ==> There are no nodes with primary role. Assuming the primary role...                                                                      │
│ postgresql postgresql-repmgr 12:53:59.54 INFO  ==> Preparing PostgreSQL configuration...                                                                                                   │
│ postgresql postgresql-repmgr 12:53:59.56 DEBUG ==> Injecting a new postgresql.conf file...                                                                                                 │
│ postgresql postgresql-repmgr 12:53:59.56 INFO  ==> postgresql.conf file not detected. Generating it...                                                                                     │
│ postgresql cp: cannot create regular file '/bitnami/postgresql/conf/postgresql.conf': Permission denied 
```

**Comment by juan131** [comment]
Thanks for providing all this useful information, I really appreciate it!

Could you please give a try to the PR I just sent trying to fix this? Thanks in advance.

**Pull request merged**